### PR TITLE
Update only affected rows in KV storage

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Update only affected rows in KV storages (EmbeddedRocksDB and KeeperMap).

cc @vdimir maybe you will be interested

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Interpreters/MutationsInterpreter.cpp [hunk 1]
```diff
@@ -697,6 +697,9 @@ void MutationsInterpreter::prepare(bool dry_run)
                     type_literal);
 
                 stages.back().column_to_updated.emplace(column, updated_column);
+
+                if (condition && return_mutated_rows)
+                    stages.back().filters.push_back(condition);
```
Comment (vdimir) [inline src/Interpreters/MutationsInterpreter.cpp:702]: How `ALTER UPDATE WHERE` worked before if we didn't apply filters?

Comment (antonio2368) [reply src/Interpreters/MutationsInterpreter.cpp:702]: It correctly mutates the rows it needs by creating `SELECT if(condition, mutate, old_value)...`. The problem for KV store is that MutationIntepreter returns all the rows causing an update on ALL keys while we only need a subset we would like to override.

Comment (vdimir) [reply src/Interpreters/MutationsInterpreter.cpp:702]: Thanks. So, for rocksdb we will not rewrite values with the same `old_value`. And for keepermap we will also not update `_version`.

**Review by vdimir — APPROVED** [review]

File: src/Interpreters/MutationsInterpreter.cpp (hunk 1 above)
Comment (antonio2368) [reply src/Interpreters/MutationsInterpreter.cpp:702]: exactly 😄

**Pull request merged**